storage/src/nodestore/alloc.rs

```diff
@@ -666,12 +706,49 @@ mod test_free_list_iterator {
         area_size_index: AreaIndex,
         offset: u64,
     ) {
-        let area: Area<Node, FreeArea> = Area::Free(FreeArea::new(next_free_block));
-        let stored_area = StoredArea::new(area_size_index, area);
-        let stored_area_bytes = serializer().serialize(&stored_area).unwrap();
+        let mut stored_area_bytes = Vec::with_capacity(16);
+        FreeArea::new(next_free_block).as_bytes(area_size_index, &mut stored_area_bytes);
         storage.write(offset, &stored_area_bytes).unwrap();
     }
 
+    #[test]
+    fn test_free_list_decoding_old_format() {
+        // old equivalent of
+        // let node = StoredArea::new(
+        //     12,
+        //     Area::<Node, _>::Free(FreeArea::new(LinearAddress::new(42))),
+        // );
+        // but encoded with bincode with varint enabled.
+        let data = [0x0c, 0x01, 0x01, 0x2a];
+
+        let mut reader = &data[..];
+
+        let area_index = reader.read_byte().unwrap();
+        assert_eq!(area_index, 12);
+
+        let free_area: FreeArea = reader.next_value().unwrap();
+        assert_eq!(
+            free_area.next_free_block(),
+            Some(LinearAddress::new(42).unwrap())
+        );
```

Comment:
This seems like this reimplementing the code in `from_reader`. Is there something you're testing here that won't get caught by running this through `from_reader`?